### PR TITLE
Issue 1 user can create an entry completed -Carolyne Golias

### PR DESCRIPTION
**Pull request by GCarolyne**

### CODE REVIEW

File: ts/data.ts [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
+  entryId: 1,
 };
+
+data += readData();
+
+function writeData(): void {
+  const dataJSON = JSON.stringify(data);
+  localStorage.setItem('data', dataJSON);
+}
+writeData();
```
Comment (shawnkost) [inline ts/data.ts:15]: Delete this, we only call this whenever we update the `data` object

File: ts/main.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+interface FormElement extends HTMLFormControlsCollection {
+  placeholder: HTMLImageElement;
+  title: HTMLInputElement;
+  photourl: HTMLInputElement;
+}
+
+interface myObjectForms {
+  title: any;
+  photourl: any;
```
Comment (shawnkost) [inline ts/main.ts:23]: We want to avoid using `any`, it is bad practice. What data types are these going to be?

File: ts/main.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  entryId: number;
+}
+
+const $form = document.querySelector('form') as HTMLFormElement;
+if (!$form) throw new Error('$form failed to query');
+
+const $data = document.querySelector('.hidden');
+if (!$data) throw new Error('$data failed to query');
```
Comment (shawnkost) [inline ts/main.ts:31]: Not needed. This isn't doing anything

File: ts/main.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+function submitForm(event: Event): void {
+  event.preventDefault();
+  const $formValues = $form.elements as FormElement;
+  const myObjectForm: myObjectForms = {
+    title: $formValues.title.value,
+    photourl: $formValues.photourl.value,
+    entryId: data.nextEntryId,
+  };
+  if (!$data) throw new Error('$data failed to query');
```
Comment (shawnkost) [inline ts/main.ts:41]: Delete this

**Review by shawnkost — CHANGES_REQUESTED** [review]
Nicely done Carolyne! Just a few changes requested here

**Review by shawnkost — APPROVED** [review]
👍🏻

**Comment by GCarolyne** [comment]
Kapture 2024-11-13 at 15 30 42

**Pull request merged**